core/common/src/test/scala/org/typelevel/otel4s/MiniMap.scala

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright 2022 Typelevel
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.typelevel.otel4s
+
+import cats.laws.discipline.ExhaustiveCheck
+import cats.syntax.all._
+import scala.collection.AbstractMap
+import scala.collection.Map
+import scala.collection.immutable
+import scala.collection.generic.IsMap
+
+// A Map with a cardinality of 25, which is a nice fit for ExhaustiveCheck.
+case class MiniMap(underlying: immutable.Map[Boolean, Compass])
+    extends AbstractMap[String, String] {
+  def iterator =
+    underlying.iterator.map { case (k, v) => k.toString -> v.toString }
+  def get(key: String) =
+    underlying.get(stringToBoolean(key)).map(_.toString)
+  def -(key1: String, key2: String, keys: String*): MiniMap =
+    keys.foldLeft(this - key1 - key2) { case (m, k) => m - k }
+  def -(key: String): MiniMap =
+    MiniMap(underlying.removed(stringToBoolean(key)))
+  private def stringToBoolean(s: String) =
+    s.toBoolean
+}
+
+object MiniMap {
+  implicit val exhaustiveCheckForMiniMap: ExhaustiveCheck[MiniMap] =
+    ExhaustiveCheck.instance(
+      ExhaustiveCheck
+        .forSet[Boolean]
+        .allValues
+        .map(_.toList)
+        .flatMap(
+          _.map(key =>
+            ExhaustiveCheck[Compass].allValues
+              .map(value => key -> value)
+          ).sequence
+        )
+        .map(_.toMap)
+        .map(MiniMap(_))
+    )
+
+  implicit val isMapForMiniMap
+      : IsMap[MiniMap] { type K = String; type V = String } =
+    new IsMap[MiniMap] {
+      type K = String
+      type V = String
+      type C = Map[String, String]
+      def apply(mm: MiniMap) = mm
+    }
+}
```

Comment:
I don't think you need this. it extends `collection.AbstractMap`, so it should already have an instance provided by the stdlib

Reply:
I think I had to be explicit about C.  I wonder if it points to a usability issue, because I expected it to just work.  This should be revisited.

Reply:
that's weird. I'll look into it

Reply:
Your other property test is encouraging for the usability.  If it can find all those, I'm not so worried about awkward test implementations like this.

Reply:
right, I've figured it out. it's because `IsMap#mapOpsIsMap` assumes some `CC[K, V] =:= C`, which is not the case here. `MiniMap` does not take parameters, and operations that change the type of its keys or values are unable to return a `MiniMap`. thus, it needs its own `IsMap` instance, just as `IntMap` and `LongMap` do. I believe you've chosen the correct path here.

I don't think this will likely be much of an issue in practice, as most collections are fully generic